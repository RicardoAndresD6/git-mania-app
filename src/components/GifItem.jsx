
import { PropTypes } from 'prop-types';

export const GifItem = ({title, url, id}) => {

    return (
        <div className="card">
            <img src={url} alt={title} className="img-gif" />
            <p className="mb-2 bg-transparent block font-sans text-xl font-semibold leading-snug tracking-normal text-blue-gray-900 antialiased">{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}