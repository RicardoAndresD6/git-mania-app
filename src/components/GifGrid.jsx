import PropTypes from "prop-types";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({category}) => {

    const {images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3 className="mt-4 text-4xl tracking-tight font-extrabold text-gray-900 sm:mt-5 sm:leading-none lg:mt-6 lg:text-5xl xl:text-6xl mb-3">{category}</h3>

            {
                isLoading && (<h2>Cargando...</h2>) 
            }

            <div className="card-grid">
                {
                    images.map( (images) => (
                        <GifItem 
                            key={images.id}
                            {...images}
                        
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}