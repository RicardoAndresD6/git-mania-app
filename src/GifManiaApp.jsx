import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifManiaApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = ( newCategory ) => {  

        if(categories.includes(newCategory )) return;

        setCategories( [ newCategory, ...categories] );
    }

    return (
        <>
            <h1 className='mt-4 text-4xl tracking-tight font-extrabold text-gray-900 sm:mt-5 sm:leading-none lg:mt-6 lg:text-5xl xl:text-6xl text-center'>GifManiaApp</h1>

            <AddCategory 
                onNewCategory={ (event => onAddCategory(event)) }
            />

            {
                categories.map( category => (
                    <GifGrid key={category} category={category} />
                ))
            }
        </>
    )
}

