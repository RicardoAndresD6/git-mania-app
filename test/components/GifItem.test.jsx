import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Pruebas en <GifItem />" , () => {

    const title = "Saitama";
    const url = "Url";

    test("Debe de hacer match con el snapshot" , () => {

        const { container }  = render( <GifItem title={title} url={url}  /> )
        expect(container).toMatchSnapshot();
    });

    test("Debe mostrar la imagen con el url y el alt aplicado", () =>{

        render( <GifItem title={title} url={url}  /> )

        const { src , alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(url);
        // screen.debug();
    });

    test("Debe mostrar el titulo del componente", () =>{

        render( <GifItem title={title} url={url}  /> )
        expect( screen.getByText('title')).toBeTruthy();
    
    });



});