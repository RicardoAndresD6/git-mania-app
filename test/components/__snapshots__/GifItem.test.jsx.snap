// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      class="img-gif"
      src="Url"
    />
    <p
      class="mb-2 bg-transparent block font-sans text-xl font-semibold leading-snug tracking-normal text-blue-gray-900 antialiased"
    >
      Saitama
    </p>
  </div>
</div>
`;
